db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -3639,4 +3620,91 @@ public double getDefaultSelectivityFactor() {
     public String getClientIPAddress() {
         return ipAddress;
     }
+
+    @Override
+    public void logStartCompiling(String statement) {
+        stmtLogger.info(String.format("Begin compiling prepared statement. %s, %s",
+                getLogHeader(), formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logEndCompiling(String statement, long nanoTimeSpent) {
+        stmtLogger.info(String.format("End compiling prepared statement. %s, timeSpent=%dms, %s",
+                getLogHeader(), nanoTimeSpent / 1000000, formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logErrorCompiling(String statement, Throwable t, long nanoTimeSpent) {
+        stmtLogger.warn(String.format("Error compiling prepared statement. %s, timeSpent=%dms, %s",
+                getLogHeader(), nanoTimeSpent / 1000000, formatLogStmt(statement)), t);
+    }
+
+    @Override
+    public void logCommit() {
+        stmtLogger.info(String.format("Committing. %s", getLogHeader()));
+    }
+
+    @Override
+    public void logRollback() {
+        stmtLogger.info(String.format("Rolling back. %s", getLogHeader()));
+    }
+
+    @Override
+    public void logStartFetching(String statement) {
+        stmtLogger.info(String.format("Start fetching from the result set. %s, %s",
+                getLogHeader(), formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logEndFetching(String statement, long fetchedRows) {
+        stmtLogger.info(String.format("End fetching from the result set. %s, fetchedRows=%d, %s",
+                getLogHeader(), fetchedRows, formatLogStmt(statement)));
+    }
+
+    @Override
+    public void logStartExecuting(String uuid, String engine, ExecPreparedStatement ps,
+                                  ParameterValueSet pvs) {
+        stmtLogger.info(String.format(
+                "Start executing query. %s, uuid=%s, engine=%s, %s, paramsCount=%d, params=[ %s ]",
+                getLogHeader(), uuid, engine, formatLogStmt(ps.getSource()),
+                pvs.getParameterCount(), pvs.toString()));
+    }
+
+    @Override
+    public void logEndExecuting(String uuid, long modifiedRows, long badRecords, long
+            nanoTimeSpent) {
+        stmtLogger.info(String.format("End executing query. %s, uuid=%s, timeSpent=%dms, " +
+                        "modifiedRows=%d, badRecords=%d",
+                getLogHeader(), uuid, nanoTimeSpent / 1000000, modifiedRows, badRecords));
+    }
+
+    private String getLogHeader() {
+        return String.format(
+                "XID=%s, SessionID=%s, Database=%s, DRDAID=%s, UserID=%s",
+                getTransactionExecute().getActiveStateTxIdString(),
+                getInstanceNumber(),
+                getDbname(),
+                getDrdaID(),
+                getSessionUserId());
+    }
+
+    private String formatLogStmt(String statement) {
+        if (statement == null) {
```

Comment:
Can you cache the formatted statement? we are calling this function over and over for logging purposes